Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2679 @@
+        BlockDeviceMappings = {'DeviceName': args.get('deviceName')}
+        BlockDeviceMappings.update({'Ebs': {}})
+    if args.get('ebsVolumeSize') is not None:
+        BlockDeviceMappings['Ebs'].update({'VolumeSize': int(args.get('ebsVolumeSize'))})
+    if args.get('ebsVolumeType') is not None:
+        BlockDeviceMappings['Ebs'].update({'VolumeType': args.get('ebsVolumeType')})
+    if args.get('ebsIops') is not None:
+        BlockDeviceMappings['Ebs'].update({'Iops': int(args.get('ebsIops'))})
+    if args.get('ebsDeleteOnTermination') is not None:
+        BlockDeviceMappings['Ebs'].update(
+            {'DeleteOnTermination': True if args.get('ebsDeleteOnTermination') == 'True' else False})
+    if args.get('ebsKmsKeyId') is not None:
+        BlockDeviceMappings['Ebs'].update({'KmsKeyId': args.get('ebsKmsKeyId')})
+    if args.get('ebsSnapshotId') is not None:
+        BlockDeviceMappings['Ebs'].update({'SnapshotId': args.get('ebsSnapshotId')})
+    if args.get('ebsEncrypted') is not None:
+        BlockDeviceMappings['Ebs'].update({'Encrypted': True if args.get('ebsEncrypted') == 'True' else False})
+    if BlockDeviceMappings:
+        kwargs.update({'BlockDeviceMappings': [BlockDeviceMappings]})  # type: ignore
+
+    if args.get('iamInstanceProfileArn') is not None:
+        kwargs.update({
+            'IamInstanceProfile': {
+                'Arn': args.get('iamInstanceProfileArn')}
+        })
+    if args.get('iamInstanceProfileName') is not None:
+        kwargs.update({  # type: ignore
+            'IamInstanceProfile': {
+                'Name': args.get('iamInstanceProfileName')}
+        })
+    if args.get('launchTemplateId') is not None:
+        kwargs.update({
+            'LaunchTemplate': {
+                'LaunchTemplateId': args.get('launchTemplateId')}
+        })
+    if args.get('launchTemplateName') is not None:
+        kwargs.update({
+            'LaunchTemplate': {
+                'LaunchTemplateName': args.get('launchTemplateName')}
+        })
+    if args.get('launchTemplateVersion') is not None:
+        kwargs['LaunchTemplate'].update({  # type: ignore
+            'Version': args.get('launchTemplateVersion')
+        })
+    if args.get('tags') is not None:
+        kwargs.update({
+            'TagSpecifications': [{
+                'ResourceType': 'instance',
+                'Tags': parse_tag_field(args.get('tags'))}]
+        })
+
+    response = client.run_instances(**kwargs)
+    data = []
+    for i, instance in enumerate(response['Instances']):
+        data.append({
+            'InstanceId': instance['InstanceId'],
+            'ImageId': instance['ImageId'],
+            'State': instance['State']['Name'],
+            'PublicIPAddress': instance.get('PublicIpAddress'),
+            'Region': obj['_user_provided_options']['region_name'],
+            'Type': instance['InstanceType'],
+            'LaunchDate': datetime.strftime(instance['LaunchTime'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
this can throw value error. better calculate it before the append and in a try block.